src/Core/System/SystemConfig/SystemConfigService.php

```diff
@@ -88,10 +93,16 @@ public function get(string $key, ?string $salesChannelId = null)
                 continue;
             }
 
-            return null;
+            $event = new SystemConfigLoadedEvent($key, null, $salesChannelId);
+            $this->eventDispatcher->dispatch($event);
+
+            return $event->getValue();
         }
 
-        return $pointer;
+        $event = new SystemConfigLoadedEvent($key, $pointer, $salesChannelId);
```

Comment:
I haven't added the event to the `all` function yet because it is a bit tricky to trigger it per value there. However, I don't think it is actually needed for all as that method isn't really used (except in tests) anyway.